sandbox/grist/imports/import_csv.py

```diff
@@ -90,108 +90,117 @@ def parse_file(file_path, parse_options=None):
-    table_name = None
-    sample_rows = list(row_set.sample)
-    # Messytables doesn't guess whether headers are present, so we need to step in.
-    data_offset, headers = import_utils.headers_guess(sample_rows)
-
-    # Make sure all header values are strings.
-    for i, header in enumerate(headers):
-      if not isinstance(header, six.string_types):
-        headers[i] = six.text_type(header)
-
-    log.info("Guessed data_offset as %s", data_offset)
-    log.info("Guessed headers as: %s", headers)
-
-    have_guessed_headers = any(headers)
-    include_col_names_as_headers = parse_options.get('include_col_names_as_headers',
-                                                     have_guessed_headers)
-
-    if include_col_names_as_headers and not have_guessed_headers:
-      # use first line as headers
-      data_offset, first_row = import_utils.find_first_non_empty_row(sample_rows)
-      headers = import_utils.expand_headers(first_row, data_offset, sample_rows)
-
-    elif not include_col_names_as_headers and have_guessed_headers:
-      # move guessed headers to data
-      data_offset -= 1
-      headers = [''] * len(headers)
-
-    row_set.register_processor(messytables.offset_processor(data_offset))
-    rows = [
-      [cell.value for cell in row]
-      for row in row_set
-    ]
-    table_data_with_types = parse_data.get_table_data(rows, len(headers), num_rows)
-
-    # Identify and remove empty columns, and populate separate metadata and data lists.
-    column_metadata = []
-    table_data = []
-    for col_data, header in zip(table_data_with_types, headers):
-      if not header and all(val == "" for val in col_data["data"]):
-        continue # empty column
-      data = col_data.pop("data")
-      col_data["id"] = header
-      column_metadata.append(col_data)
-      table_data.append(data)
-
-    if not table_data:
-      # Don't add tables with no columns.
-      continue
-
-    guessed = row_set._dialect
-    quoting = parse_options.get('quoting')
-    options = {"delimiter": parse_options.get('delimiter', guessed.delimiter),
-               "doublequote": parse_options.get('doublequote', guessed.doublequote),
-               "lineterminator": parse_options.get('lineterminator', guessed.lineterminator),
-               "quotechar": parse_options.get('quotechar', guessed.quotechar),
-               "skipinitialspace": parse_options.get('skipinitialspace', guessed.skipinitialspace),
-               "include_col_names_as_headers": include_col_names_as_headers,
-               "start_with_row": 1,
-               "NUM_ROWS": num_rows,
-               "SCHEMA": SCHEMA
-               }
-
-    log.info("Output table %r with %d columns", table_name, len(column_metadata))
-    for c in column_metadata:
-      log.debug("Output column %s", c)
-    export_list.append({
-      "table_name": table_name,
-      "column_metadata": column_metadata,
-      "table_data": table_data
-    })
+
+  table_name = None
```

Comment:
Please move `table_name` and `num_rows` down to where they're used. I know you didn't touch them, but moving the lines you changed instead would make less sense.

Reply:
Looking at this, it seems that table_name is never set. Or am I missing something ?

Reply:
Yes, I think `table_name` is for the other file types which can have multiple tables each with their own name, e.g. the sheet name in Excel. The way the code suggested multiple tables were possible with a loop is weird.

Reply:
Should I then drop the variable ?

Reply:
Yes, I think dropping it would be clearer.